tests/python/pants_test/backend/python/tasks/native/test_ctypes.py

```diff
@@ -90,39 +91,80 @@ def run_before_task_types(cls):
-    self.assertEqual(['platform_specific_ctypes_cpp_dist==0.0.0+{}'.format(snapshot_version)],
-                      [str(x.requirement) for x in synthetic_target.requirements.value])
-
-    local_wheel_products = context.products.get('local_wheels')
-    local_wheel = self.retrieve_single_product_at_target_base(
-      local_wheel_products, platform_specific_dist)
-    self.assertTrue(check_wheel_platform_matches_host(local_wheel))
+    self._assert_dist_and_wheel_identity(
+      'platform_specific_ctypes_cpp_dist', '0.0.0', self.ExpectedPlatformType.current,
+      platform_specific_dist, extra_targets=[self.target_dict['ctypes_cpp_library']],
+    )
+
+  def test_multiplatform_python_setup_resolve_bypasses_python_setup(self):
+    self.set_options_for_scope('python-setup',
+                               platforms=['current', 'linux-x86_64', 'macosx_10_14_x86_64'])
+    platform_specific_dist = self.target_dict['platform_specific_ctypes_cpp_dist']
+    self._assert_dist_and_wheel_identity(
+      'platform_specific_ctypes_cpp_dist', '0.0.0', self.ExpectedPlatformType.current,
+      platform_specific_dist,
+      extra_targets=[self.target_dict['ctypes_cpp_library']],
+    )
+
+  def test_resolve_for_native_sources_allows_current_platform_only(self):
+    platform_specific_dist = self.target_dict['platform_specific_ctypes_cpp_dist']
+    compatible_python_binary_target = self.make_target(
+      spec='src/python/plat_specific:bin',
+      target_type=PythonBinary,
+      dependencies=[platform_specific_dist],
+      entry_point='this-will-not-run',
+      platforms=['current'],
+    )
+    self._assert_dist_and_wheel_identity(
+      'platform_specific_ctypes_cpp_dist', '0.0.0', self.ExpectedPlatformType.current,
+      platform_specific_dist,
+      extra_targets=[
+        self.target_dict['ctypes_cpp_library'],
+        compatible_python_binary_target,
+      ])
+
+  def test_multiplatform_resolve_with_binary(self):
+    platform_specific_dist = self.target_dict['platform_specific_ctypes_cpp_dist']
+    incompatible_python_binary_target = self.make_target(
+      spec='src/python/plat_specific:bin',
+      target_type=PythonBinary,
+      dependencies=[platform_specific_dist],
+      entry_point='this-will-not-run',
+      platforms=['current', 'linux-x86_64', 'macosx_10_14_x86_64'],
+    )
+    with self.assertRaisesWithMessage(IncompatiblePlatformsError, dedent("""\
+      The target set contains one or more targets that depend on native code. Please ensure that the
+      platform arguments in python_binary() targets, as well as the value of
+      --python-setup-platforms, are compatible with the current platform.
+      Platform assignments for python targets: {'current': OrderedSet([PythonBinary(src/python/plat_specific:bin)]), 'linux-x86_64': OrderedSet([PythonBinary(src/python/plat_specific:bin)]), 'macosx_10_14_x86_64': OrderedSet([PythonBinary(src/python/plat_specific:bin)])}
```

Comment:
The dict + OrderedSet + PythonBinary representation here isn't very user-friendly. Ideally this would read more like:
`Target src/python/plat_specific:bin has unsupported platforms linux-x86_64 and macosx_10_14_x86_64`

(and ideally we would resolve current before doing this check, so that we know that current == linux or macosx)

Reply:
```python
from pex import pep425tags
print(pep425tags.get_platform())
```
is how to resolve the current platform